tests/func/test_repro.py

```diff
@@ -715,6 +715,8 @@ def test(self):
         stages = self.dvc.reproduce(target)
         self.assertEqual(len(stages), 0)
 
+        self.dvc.odb.local.unprotect(self.DATA_SUB)
+
```

Comment:
Btw, were you able to figure out why this got triggered?

Reply:
Indeed: I wasn't honoring `dry_run` for `stage()`. Fixed now and got rid of `unprotect` in this test.